src/machines/auth-token-polling-machine.ts

```diff
@@ -0,0 +1,100 @@
+import {createMachine, assign} from 'xstate'
+
+type DoneEventObject = import('xstate').DoneEventObject
+
+type Event = DoneEventObject
+
+interface Context {
+  stripeCheckoutSessionId?: string
+  authToken?: string
+  pollingCount: number
+}
+
+const POLLING_LIMIT = 5
+
+export const authTokenPollingMachine = createMachine<Context, Event>(
+  {
+    initial: 'pending',
+    context: {
+      stripeCheckoutSessionId: undefined,
+      authToken: undefined,
+      pollingCount: 0,
+    },
+    states: {
+      pending: {
+        initial: 'polling',
+        states: {
+          polling: {
+            entry: 'increasePollingCount',
+            invoke: {
+              src: 'requestAuthToken',
+              onDone: {
+                target: 'verifyRetrieval',
+                actions: ['assignAuthToken'],
+              },
+              onError: {
+                target: 'scheduleNextPoll',
+              },
+            },
```

Comment:
This is machine doesn't get sent any events. Instead, it starts in the `{pending: "polling"}` state which invokes the `requestAuthToken` service.

From there it will either:
- `onError` --> schedule the next poll after 2000ms, or transition to a final failed state.
- `onDone` --> check the auth token was set, apply any more delay to keep UI smooth, and then transition to a final state.